### ISSUE TITLE
`file://` URLs are not vaild, resulting in a warning

### ISSUE THREAD

**Issue by eirnym** [huhu/rust-search-extension#290]
it's unfortunate to see ` Error: URL constructor:  is not a valid URL.` every time I open a local file. if `URL` can't be constructed with this scheme, please hide warning, it's really annoying

**Comment by Folyd** [comment]
Hi thanks for the feedback. Can you give a screenshot? Which platform do you use?

**Comment by eirnym** [comment]
I use Firefox on macOS if OS important. I found that Chromium-based browsers don't show most warnings, providing false sense of correctness.

**Comment by eirnym** [comment]
more information which could help:

While having style editor of an offline page (`file://` protocol) open, I've opened a console below (!), style editor and inside styles I've introduced a mistake, which triggered the error, as you can see on a screenshot. 

I _guess_ source of the error is following: plugin parses everything on a screen  and tries to guess if this a proper page to be applied to, and it parses devtools window as well. 

<img width="1033" alt="Zrzut ekranu 2024-09-29 o 17 45 08" src="https://github.com/user-attachments/assets/6f296260-6947-40d7-8826-941a046b7106">

**Comment by Folyd** [comment]
Yes, Firefox doesn't allow `file://` protocol, please see the FAQ: https://rust.extension.sh/faq/#why-local-file-rust-doc-not-work-properly-on-firefox

**Comment by eirnym** [comment]
Please, reopen it. I'd like warning to be removed entirely
